include/deal.II/lac/affine_constraints.h

```diff
@@ -2309,14 +2309,6 @@ struct IsBlockMatrix
   static std::true_type
   check(const BlockMatrixBase<T> *);
 
-  /**
-   * Overload returning true if the class is derived from
-   * BlockSparsityPatternBase, which is what block sparsity patterns do.
-   */
-  template <typename T>
-  static std::true_type
-  check(const BlockSparsityPatternBase<T> *);
```

Comment:
You could rather deprecate this.

Reply:
It's a private member function, so there is no point in deprecating it.

I think that this function is actually in the `dealii` namespace by accident. It's only used in 4 repeated places where we dispatch to different functions. We would likely put it into an internal namespace today, and that was my plan for a follow-up patch. It is true that this PR changes the semantics of the class (it was poorly named because it does two things at the same time, and I'm breaking it up into two classes). One could argue that I should deprecate the old class, but like I said above: I think the class was where it is by accident, and I can't imagine that it was actually used anywhere outside that file.